src/ipa/rpi/pisp/pisp.cpp

```diff
@@ -112,6 +113,27 @@ int generateLut(const ipa::Pwl &pwl, uint32_t *lut, std::size_t lutSize,
 	return 0;
 }
 
+int generateDecompandLut(const ipa::Pwl &pwl, uint16_t *lut, std::size_t lutSize = 65)
+{
+	if (pwl.empty())
+		return -EINVAL;
+
+	constexpr int step = 1024;
+	for (std::size_t i = 0; i < lutSize; ++i) {
+		int x = i * step;
+		if (x > 65535)
```

Comment:
The _final_ control point (only) is allowed to be at (65536, 65536) so strictly speaking we don't need this clip, nor the `min` at line 131. However the difference is so tiny that it really doesn't matter.